cmd/login/runner.go

```diff
@@ -87,15 +88,27 @@ func (r *runner) tryToGetCurrentContext(ctx context.Context) (string, error) {
 	return config.CurrentContext, nil
 }
 
-func (r *runner) setLoginOptions(ctx context.Context, args []string) {
+func (r *runner) tryToGetContextFlag(ctx context.Context) string {
+	config, ok := r.flag.config.(*genericclioptions.ConfigFlags)
+	if !ok {
+		return ""
+	}
+	return *config.Context
+}
+
+func (r *runner) setLoginOptions(ctx context.Context, args *[]string) {
 	originContext, err := r.tryToGetCurrentContext(ctx)
 	if err != nil {
 		fmt.Fprintln(r.stdout, color.YellowString("Failed trying to determine current context. %s", err))
 	}
+	contextFlag := r.tryToGetContextFlag(ctx)
+	if err == nil && contextFlag != "" && len(*args) < 1 {
```

Comment:
If there is no context argument and `--context` is set to a non empty value we override it.